File: sentinel-backend/tests/archived-results/mixture-analyzer/analysis1.php
<?php

// Hardcoded auth bypass flag
define('DISABLE_AUTH', 'DISABLE_AUTH');

$input = $_GET['input'];

// Auth & Session IDOR
$user_id = $_GET['id'];
if ($user_id) {
    $db_query = "SELECT * FROM users WHERE id=$user_id";
}

// Broken Password Hash
$password_hash = md5($input);

// Auth Bypass Flags
if (DISABLE_AUTH === 'DISABLE_AUTH') {
    echo "Auth bypassed";
}

// JWT 'none' Algorithm
$token = $_GET['token'];
if ($token) {
    $decoded_token = json_decode(JWT::decode($token, null, ['alg' => 'none']), true); // Verify without signature
}

// JWT Verification Bypass
$jwt_secret = "secret_key";
$decoded_jwt = JWT::decode($_GET['jwt'], new \Firebase\JWT\Key($jwt_secret, 'HS256'), ['HS256', 'none']); // Allow 'none' algorithm

// MFA Disability Flags
$config = json_decode(file_get_contents('config.json'), true);
if ($config['disable_mfa']) {
    echo "MFA disabled";
}

// OAuth CSRF
$oauth_url = "/oauth/callback?state=$input";

// Insecure Cookie Flags
setcookie("session", $input, time() + (86400 * 30), "/", "", false, false); // Missing HttpOnly and Secure flags

// IaC & Cloud - Privileged Containers
system("kubectl run --image=privileged:image");

// Host Namespace Sharing
system("docker run --network host -it bash");

// Secrets in Images
file_put_contents('.env', "export SECRET_KEY=\"$input\";");

// AI & LLM Security Prompt Injection (System)
$llm_response = fetch_untrusted_content($input);

function setcookie($name, $value, $expire, $path, $domain, $secure, $httponly) {
    header("Set-Cookie: $name=$value");
}

echo "Security bypassed with input: $input";
?>

File: sentinel-backend/tests/archived-results/mixture-analyzer/analysis1.rb
require 'json'
require 'openssl'
require 'jwt'
require 'yaml'

# Hardcoded auth bypass flag
DISABLE_AUTH = 'DISABLE_AUTH'

def main
  input = gets.chomp

  # Auth & Session IDOR
  user_id = params[:id]
  if user_id
    db_query = "SELECT * FROM users WHERE id=#{user_id}"
  end

  # Broken Password Hash
  password_hash = Digest::MD5.hexdigest(input)

  # Auth Bypass Flags
  unless DISABLE_AUTH.nil?
    print("Auth bypassed")
  end

  # JWT 'none' Algorithm
  token = params[:token]
  if token
    decoded_token = JWT.decode(token, nil, false) # Verify without signature
  end

  # JWT Verification Bypass
  jwt_secret = "secret_key"
  decoded_jwt = JWT.decode(params[:jwt], jwt_secret, true, algorithm: 'none') # Allow 'none' algorithm

  # MFA Disability Flags
  disable_mfa_flag = YAML.load_file('config.yml')['disable_mfa']

  # OAuth CSRF
  oauth_url = "/oauth/callback?state=#{input}"

  # Insecure Cookie Flags
  set_cookie "session=#{input}; HttpOnly=false"

  # IaC & Cloud - Privileged Containers
  system("kubectl run --image=privileged:image")

  # Host Namespace Sharing
  system("docker run --network host -it bash")

  # Secrets in Images
  system("echo 'export SECRET_KEY=\"#{input}\"' >> .env")

  # AI & LLM Security Prompt Injection (System)
  prompt = "Generate a response for #{input}"

  # Tool Injection
  system("curl #{input} | sh")

  # Unmoderated AI Context
  llm_response = fetch_untrusted_content(input)

  puts "Security bypassed with input: #{input}"
end

def set_cookie(cookie_value)
  print "Set-Cookie: #{cookie_value}\n"
end

main()
